setup.rst

```diff
@@ -34,22 +34,58 @@ installation directions. You may also want to consider a graphical client
 such as `TortoiseGit <https://tortoisegit.org/>`_ or
 `GitHub Desktop <https://desktop.github.com/>`_.
 
-You may also wish to set up :ref:`your name and email <set-up-name-email>`
-and `an SSH key
+Once you installed Git, you should set up
+:ref:`your name and email <set-up-name-email>` and `an SSH key
 <https://help.github.com/articles/adding-a-new-ssh-key-to-your-github-account/>`_
 as this will allow you to interact with GitHub without typing a username
 and password each time you execute a command, such as ``git pull``,
 ``git push``, or ``git fetch``.  On Windows, you should also
 :ref:`enable autocrlf <autocrlf>`.
 
+
 .. _checkout:
 
 Getting the Source Code
 -----------------------
 
-In order to get a copy of the source code you should first :ref:`fork the
-Python repository on GitHub <fork-cpython>` and then :ref:`create a local
-clone of your private fork and configure the remotes <clone-your-fork>`.
+In order to get a copy of the source code you should :ref:`fork the
+Python repository on GitHub <fork-cpython>`, :ref:`create a local
+clone of your personal fork, and configure the remotes <clone-your-fork>`.
+
+You will only need to execute these steps once:
+
+1. Go to https://github.com/python/cpython.
+
+2. Press :guilabel:`Fork` on the top right.
+
+3. When asked where to fork the repository, choose to fork it to your username.
+
+4. Your fork will be created at :samp:`https://github.com/{<username>}/cpython`.
+
+5. Clone your GitHub fork (replace ``<username>`` with your username)::
+
+      $ git clone git@github.com:<username>/cpython.git
+
+   (You can use both SSH-based or HTTPS-based URLs.)
+
+6. Configure an ``upstream`` remote::
+
+      $ cd cpython
+      $ git remote add upstream git@github.com:python/cpython.git
```

Comment:
@ezio-melotti I think this is correct if you have push privileges on the repo, but others likely will need to use https for the upstream repo.

Reply:
One has to be using ssh for git@github to work.  I believe we agree to consistently use https: (and gits credential keeper) in the instructions as using both randomly is confusing.

Reply:
But for the record, it does work even if you don't have push privileges for the repo, if you are using ssh already with github.

Reply:
This was already discussed on #263.  The outcome was that by using this URL we don't need to have separate instructions for core devs and contributors.  Pulling  works for both, and pushing to upstream would fail anyway if contributors don't have push privileges.
This was also moved from the old instructions, I didn't change it :)